File: Benchmarking/30 Euler/euler10/old/euler10.java
public class euler10
{
	public static void main(String[] args)
	{
		long limit = Long.parseLong(args[0]);
		System.out.println(sumPrime(limit));
	}
	
	private static boolean isPrime(long x)
	{
		long divisor = 2;
		boolean prime = true;
		while(divisor < x)
		{
			if(x % divisor == 0)
			{
				prime = false;
				break;
			}
			divisor += 1;
		}
		return prime;
	}
	
	private static long sumPrime(long n)
	{
		long sum = 0;
		for(long i = 2; i < n; i++)
		{
			if(isPrime(i))
			{
				sum += i;
			}
		}
		return sum;
	}
}

File: Benchmarking/30 Euler/euler10/old/euler10.jl
#=
The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.
=#
#Answer: 142913828922

#Terminal: Julia euler10.jl 2000000
limit = parse(Int, ARGS[1])
function isPrime(x)
  divisor = 2
  prime = true
  while divisor < x
    if x % divisor == 0
      #@printf "%d mod %d == 0\n" x divisor
      prime = false
      break
    end
    divisor += 1
  end
  return prime
end

function sumPrime(n)
  sum = 0
  for i=2:n
    if isPrime(i)
      sum += i
    end
  end
  return sum
end
println(sumPrime(limit))
#test primes()
#Sieve of Eratosthenes
